resqpy/well/_trajectory.py

```diff
@@ -231,7 +229,10 @@ def __choose_init_method(self, set_tangent_vectors, data_frame, cell_kji0_list,
         elif wellspec_file:
             return 'wellspec_file'
         elif deviation_survey_file:
-            return 'deviation_survey_file'
+            return 'ascii_trajectory_file'
+        elif set_tangent_vectors and type(self.knot_count) is int and self.tangent_vectors is None:
+            if len(self.knot_count) > 1:
+                return 'tangent_vectors'
 
```

Comment:
I would add a 'return None' line here to explicitly cover the situation when none of the conditions are met.  (Or raise an exception if it is an invalid situation, though here I think it should be allowed.)

Reply:
@andy-beer, I added the recommended 'return None' line to the __choose_init_method(). The try-except block  starting at line 201 should then log a warning if the combination of input arguments is invalid.